src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -299,24 +294,25 @@ NonPushableInsertSelectExecScan(CustomScanState *node)
 
 
 /*
- * TwoPhaseInsertSelectTaskList generates a list of tasks for a query that
+ * GenerateTaskListWithIntermediateResults generates a list of tasks for a query that
  * inserts into a target relation and selects from a set of co-located
  * intermediate results.
  */
-static List *
-TwoPhaseInsertSelectTaskList(Oid targetRelationId, Query *insertSelectQuery,
-							 char *resultIdPrefix)
+List *
```

Comment:
do you think can/should we move this to a new file called repartition_executor.c? Same for `GenerateTaskListWithRedistributedResults`

ps: should we call this concept as: repartition_subquery_executor.c  ? If you agree, please add some file comments on the top of the file

Reply:
I've moved both the routines to a new file `repartition_executor.c` Once this is pushed, I will do another refactoring, taking out the common stuff in merge and insert-select to new routines (if possible)